### PR TITLE
Convert storage temporaries in fv3core to Quantity

### PR DESCRIPTION
**Pull request by mcgibbon**
## Purpose

Changes to gt4py will soon allow custom classes to be used in place of storages. We'd like to take advantage of this by using Quantity everywhere in the model. This PR refactors all internal temporaries in fv3core to use Quantity.

## Code changes:

- Switched some 1-dimensional computation from stencils to pure numpy/cupy
- added fC and fC_agrid property methods to GridData, instead of computing inside dycore init methods
- Added QuantityFactory to init arguments of all classes that create temporary data
- Internal temporaries in fv3core are now Quantity instead of storage
- All storage arguments to model components in fv3core are now Quantity

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://drive.google.com/file/d/1R0nqOxfYnzaSdoYdt8yjx5J482ETI2Ft/view?usp=sharing).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] For each public change and fix in `pace-util`, HISTORY has been updated
- [x] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

File: fv3core/pace/fv3core/stencils/delnflux.py [hunk 1]
```diff
@@ -1,29 +1,41 @@
...
-        damp4 = (damp_c * da_min) ** (nord + 1)
+def calc_damp(
+    damp_c: pace.util.Quantity, da_min: float, nord: pace.util.Quantity
+) -> pace.util.Quantity:
+    if damp_c.dims != nord.dims or damp_c.data.shape != nord.data.shape:
+        raise NotImplementedError(
+            "current implementation requires damp_c and nord to have "
+            "identical data shape and dims"
+        )
+    data = (damp_c.data * da_min) ** (nord.data + 1)
```
Comment (elynnwu) [inline fv3core/pace/fv3core/stencils/delnflux.py:29]: Why is this moved away from gt4py stencil?

Comment (elynnwu) [reply fv3core/pace/fv3core/stencils/delnflux.py:29]: Are all the removed stencils because it's called once in init only?

**Review by elynnwu — COMMENTED** [review]
Do we want to do physics and stencils in a different PR?

File: fv3core/pace/fv3core/stencils/delnflux.py (hunk 1 above)
Comment (mcgibbon) [reply fv3core/pace/fv3core/stencils/delnflux.py:29]: Yes. I removed the gt4py stencils done for computing 1D values to avoid adding more complexity to the already complex make-a-3d-storage-to-compute-to-subset-into-a-1d-storage logic.

**Comment by mcgibbon** [comment]
launch jenkins

**Review by elynnwu — APPROVED** [review]
